omod-common/src/main/java/org/openmrs/module/webservices/docs/swagger/SwaggerSpecificationCreator.java

```diff
@@ -910,7 +894,65 @@ private void addPaths() {
 		Collections.sort(tagList);
 		swaggerSpecification.setTags(tagList);
 	}
-	
+
+	/**
+	 * Extracts resource name from a resource handler
+	 * @param trim if false returns resource name with version, if true version part of the name is trimmed
+	 * @return resource name
+	 */
+	private String getResourceName(DelegatingResourceHandler resourceHandler, boolean trim) {
+		Resource annotation = resourceHandler.getClass().getAnnotation(Resource.class);
+
+		String resourceName = null;
+		if (annotation != null) {
+			// top level resource
+			resourceName = annotation.name();
+		} else {
+			// sub resource
+			SubResource subResourceAnnotation = resourceHandler.getClass().getAnnotation(SubResource.class);
+			if (subResourceAnnotation != null) {
+				resourceName = subResourceAnnotation.path();
+			}
+		}
+		if (trim) {
+			resourceName = trimResourceName(resourceName);
+		}
+		return resourceName;
+	}
+
+	/**
+	 * Extracts resource parent name from a resource handler
+	 * @param trim if false returns name with version, if true version part of the name is trimmed
+	 * @return name of the parent resource or null if its not a sub-resource
+	 */
+	private String getResourceParentName(DelegatingResourceHandler resourceHandler, boolean trim) {
+		SubResource subResourceAnnotation = resourceHandler.getClass().getAnnotation(SubResource.class);
```

Comment:
Sorry I'm still confused. If we are to add a boolean like `getParent` then we have to assume that the caller knows if its SubResource or not before calling the method. I can't seem to get things to delegate without making things extra complicated. 

I wouldn't mind you making a commit or gist, if you've got the time of course! Thanks.